src/replica/duplication/replica_follower.cpp

```diff
@@ -61,8 +62,86 @@ void replica_follower::init_master_info()
     }
 }
 
+error_code replica_follower::duplicate_checkpoint()
+{
+    if (_duplicating_checkpoint) {
+        dwarn_replica("duplicate master[{}] checkpoint is running", master_replica_name());
+        return ERR_BUSY;
+    }
+
+    ddebug_replica("start duplicate master[{}] checkpoint", master_replica_name());
+    zauto_lock l(_lock);
+    _duplicating_checkpoint = true;
+    async_duplicate_checkpoint_from_master_replica();
+    _tracker.wait_outstanding_tasks();
+    _duplicating_checkpoint = false;
+    return _tracker.result();
+}
+
+void replica_follower::async_duplicate_checkpoint_from_master_replica()
+{
+    rpc_address meta_servers;
+    meta_servers.assign_group(_master_cluster_name.c_str());
+    meta_servers.group_address()->add_list(_master_meta_list);
+
+    configuration_query_by_index_request meta_config_request;
+    meta_config_request.app_name = _master_app_name;
+    meta_config_request.partition_indices = {get_gpid().get_partition_index()};
+
+    ddebug_replica("query master[{}] replica configuration", master_replica_name());
+    dsn::message_ex *msg = dsn::message_ex::create_request(
+        RPC_CM_QUERY_PARTITION_CONFIG_BY_INDEX, 0, get_gpid().thread_hash());
+    dsn::marshall(msg, meta_config_request);
+    rpc::call(meta_servers,
+              msg,
+              &_tracker,
+              [&](error_code err, configuration_query_by_index_response &&resp) mutable {
+                  tasking::enqueue(LPC_DUPLICATE_CHECKPOINT, &_tracker, [=]() mutable {
+                      FAIL_POINT_INJECT_F("duplicate_checkpoint_ok", [&](string_view s) -> void {
+                          _tracker.set_result(ERR_OK);
```

Comment:
I consider it is strange to pass the copy-duplication-checkpoint result through tracker, especially in this pr, it is only used in unit test.
How about using a local variety, like _checkpoint_result?

Reply:
- it's not just used for `unit test`,  duplicate_checkpoint of [caller](https://github.com/XiaoMi/rdsn/blob/duplication_dev/src/replica/replica_init.cpp#L91) will use the result
- tracker of `rdsn`  has the shortcoming, when we want get some `async work` result, we had to define `local variable` to update the result in every time we use it, I don't think it's one good usage. define `result` in `tracker` can help us track and get async work result,  avoid define one local variable after another

Reply:
- My meaning is that the result seems will always be ERR_OK, will it be a error only in test case?
- If the tracker only has one task, it sounds a good solution. Actually, task_tacker has a task list, it is difficult for you to track which result belonging to which task. Besides, as my first comment, will the `result` be other error code rather than `ERR_OK`?

Reply:
As we discuss offline, I recommend using `bool` instead of `error_code`.